service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -76,9 +76,12 @@ fi
 # Log in with app-default-credentials
 sudo -u "${JUPYTER_USER}" sh -c "terra auth login --mode=APP_DEFAULT_CREDENTIALS"
 
-# Get all the git repvoio clone urls from the workspace resource list with type GIT_REPO and try to clone all of them.
+sudo -u "${JUPYTER_USER}" sh -c "mkdir -p /home/${JUPYTER_USER}/.ssh"
+cd /home/${JUPYTER_USER}/.ssh
+sudo -u "${JUPYTER_USER}" sh -c "terra user ssh-key get --format=JSON | jq -r '.privateSshKey'> id_rsa"
```

Comment:
What happens if the user hasn't already registered an SSH key? I don't know how notebooks handle failure in startup scripts. 

also, the previous approach allowed users to clone public repos without setting up an ssh key with Github, but it seems like that wouldn't happen now. Is that behavior we care about, or was that just useful before we had ssh keys?

Reply:
+1, I think we still need to try the repos below even if the key fails.  and likely ensure we try them all such that we can clone a public one even if a private one fails?

Reply:
good point. done